api/core/tools/tool_engine.py

```diff
@@ -78,8 +81,27 @@ def agent_invoke(tool: Tool, tool_parameters: Union[str, dict],
                     logging.error(f"Traceback {traceback.format_exc()}")
                     raise e
             else:
-                logging.info(f"INVOKING TOOL FOR {request_info}.")
+                posthog.capture(
+                    request_info['username'],
+                    'tool_invoked',
+                    {
+                        "conversation_id": message.conversation_id,
+                        "tool_name": tool.identity.name,
+                        "tool_inputs": json.dumps(tool_parameters) if isinstance(tool_parameters, dict) else tool_parameters
+                    }
+                )
                 meta, response = ToolEngine._invoke(tool, tool_parameters, user_id)
```

Comment:
i think if chat spawner is called this will not execute. pls double check implementation above in the agent tool invocation method

Reply:
That's correct. I don't think an event should be logged when chat spawner is called. The tool_invoked event should be logged only when a tool is called

Reply:
agreed, its not necessary